41-vaccine-temp-monitor/firmware/arduino/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+        JAddNumberToObject(cmd, "sensitivity", 1);
+        notecard.sendRequest(cmd);
+    }
+
+    // Configure GPS heartbeat to verify last location reported
+    if (J *cmd = notecard.newCommand("card.location.track"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddBoolToObject(cmd, "sync", true);
+        JAddBoolToObject(cmd, "heartbeat", true);
+        JAddNumberToObject(cmd, "hours", 12);
+        notecard.sendRequest(cmd);
+    }
+
+    // Establish a template to optimize queue size and data usage
+    if (J *cmd = notecard.newCommand("note.template"))
+    {
+        JAddStringToObject(cmd, "file", "status.qo");
+        if (J *body = JAddObjectToObject(cmd, "body"))
+        {
+            JAddBoolToObject(body, "alert", TBOOL);
+            JAddBoolToObject(body, "low_batt", TBOOL);
+            JAddNumberToObject(body, "temp", TFLOAT16);
+            JAddNumberToObject(body, "voltage", TFLOAT16);
+            notecard.sendRequest(cmd);
+        }
+    }
+}
+
+bool currentlyMoving(void)
+{
+    bool moving;
+    uint32_t movementTimeSecs = 0;
+
+    // Increase motion sensitivity to help detect movement
+    if (J *cmd = notecard.newCommand("card.motion.mode"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddNumberToObject(cmd, "sensitivity", 5);
+        notecard.sendRequest(cmd);
+    }
+
+    // Determine whether or not the module is moving.
+    J *rsp = _notecard.requestAndResponse(_notecard.newRequest("card.triangulate"));
+    if (!rsp)
+    {
+        moving = false; // prioritize battery life on error
+    }
+    else
+    {
+        uint32_t movementTimeSecs = JGetInt(rsp, "motion");
+        NoteDeleteResponse(rsp);
+
+        // Wait long enough for the accelerometer to record a new timestamp.
+        ::delay(1500);
+
+        rsp = _notecard.requestAndResponse(_notecard.newRequest("card.triangulate"));
+    }
+
+    // Restore motion sensitivity to preserve battery life
+    if (J *cmd = notecard.newCommand("card.motion.mode"))
+    {
+        JAddBoolToObject(cmd, "start", true);
+        JAddNumberToObject(cmd, "sensitivity", 1);
+        notecard.sendRequest(cmd);
+    }
+
+    return valid;
+}
+
+void envVarManagerCb(const char *var, const char *val, void *userCtx)
+{
+    // Cache the values for each variable.
+    if (strcmp(var, "alarm_temperature") == 0)
+    {
+        alarm_temperature = atoi(val);
+    }
+    else if (strcmp(var, "alarm_hysteresis") == 0)
+    {
+        alarm_hysteresis = atoi(val);
+    }
+    else if (strcmp(var, "sample_interval_s") == 0)
+    {
+        sample_interval_s = atoi(val);
+    }
+}
+
+void fetchEnvironmentVariables(void)
+{
+    static const char *envVars[] = {
+        "alarm_temperature",
+        "alarm_hysteresis",
+        "sample_interval_s"};
+    static const size_t numEnvVars = sizeof(envVars) / sizeof(envVars[0]);
+
+    // Allocate the environment variable manager.
+    if (!(envVarManager = NotecardEnvVarManager_alloc()))

```

Comment:
Does all the memory get wiped out on sleep? Just trying to figure out why we can continually alloc the manager and never free it.

Reply:
Yes.

BUT... I would agree this is not a very tidy thing to do.

I'll add `NotecardEnvVarManager_free()` to clean up after myself.